tests/phpunit/integration/Core/Notifications/NotificationsTest.php

```diff
@@ -42,4 +49,44 @@ function ( REST_Route $route ) {
 			$routes
 		);
 	}
+
+	public function test_non_proxy_request_does_not_request_proxy() {
+
+		$context                        = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE, new MutableInput() );
+		$google_proxy                   = new Google_Proxy( $context );
+		$google_proxy_notifications_url = $google_proxy->url( '/notifications/' );
+
+		// Create a user with access to the WP REST API and log in.
+		$user = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user->ID );
+		do_action( 'wp_login', $user->user_login, $user );
+
+		// Fake setup and authentication for access to dashboard.
+		$this->fake_site_connection();
+
+		add_filter(
+			'pre_http_request',
+			function ( $override, $args, $url ) use ( $google_proxy_notifications_url ) {
+				if ( strstr( $url, $google_proxy_notifications_url ) ) {
+					$this->fail( 'Expected no request to be made to Google proxy notifications endpoint' );
+				}
+				return $override;
+			},
+			10,
+			3
+		);
+
+		// Get a REST server instance.
+		$server = rest_get_server();
+
+		// Make the test request.
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/site/data/notifications' );
+		$response = $server->dispatch( $request );
+
+		// Confirm the request returns 200 status code.
+		$this->assertEquals(
+			$response->get_status(),
+			200
+		);
```

Comment:
Expected and actual values are backwards here 😄 
```suggestion
		$this->assertEquals( 200, $response->get_status() );
```